drop-boxes/register-nanopore-dropbox/register-nanopore.py

```diff
@@ -0,0 +1,281 @@
+    usedExperimentIdentifiers.append(newExpID)
+    return transaction.createNewExperiment(newExpID, NANOPORE_EXP_TYPE_CODE)
+
+# return first line of file
+def getDatahandlerMetadata(incomingPath, fileName):
+    path = os.path.realpath(os.path.join(incomingPath,file_name))
+    with open(path) as f:
+        return f.readline()
+
+def getTimeStamp():
+    now = datetime.now()
+    ts = str(now.minute)+str(now.second)+str(now.microsecond)
+    return ts
+
+# return absolute path of copied folder containing all log files. used to add log files to each involved sample
+def copyLogs(parentPath, fileList):
+    ts = getTimeStamp()
+    newLogFolder = os.path.join(parentPath, ts+"/logs")
+    os.mkdir(newLogFolder)
+    for logFile in fileList:
+        src = os.path.join(parentPath, logFile)
+        shutil.copy2(src, newLogFolder)
+    return newLogFolder
+
+def handleMeasurement(transaction, space, project, measurement, origin, rawDataPerSample):
+    #reminder: incoming path is of the absolute path of the folder created by the datahandler.
+    #joining this path with any relative path returned by the nanopore object will give the absolute path of that file/folder.
+    incomingPath = transaction.getIncoming().getAbsolutePath()
+    currentPath = os.path.join(incomingPath, measurement.getRelativePath())
+    # handle metadata of experiment level
+    runExperiment = createNewExperiment(transaction, space, project)
+
+    #do we get these automatically? from where?
+    runExperiment.setPropertyValue("Q_ASIC_TEMPERATURE", measurement.getAsicTemp())
+    runExperiment.setPropertyValue("Q_NGS_BASE_CALLER", measurement.getBaseCaller()+ " " +measurement.getBaseCallerVersion())
+    runExperiment.setPropertyValue("Q_SEQUENCER_DEVICE", measurement.getDeviceType())
+    runExperiment.setPropertyValue("Q_FLOWCELL_BARCODE", measurement.getFlowcellId())
+    runExperiment.setPropertyValue("Q_FLOWCELL_POSITION", measurement.getFlowCellPosition())
+    runExperiment.setPropertyValue("Q_FLOWCELL_TYPE", measurement.getFlowCellType())
+    runExperiment.setPropertyValue("Q_LIBRARY_PREPKIT", measurement.getLibraryPreparationKit())
+    runExperiment.setPropertyValue("Q_NGS_NANOPORE_HOSTNAME", measurement.getMachineHost())
+    runExperiment.setPropertyValue("Q_DATA_GENERATION_FACILITY", origin)
+    runExperiment.setPropertyValue("Q_MEASUREMENT_START_DATE", measurement.getStartDate())
+    # runExperiment.setPropertyValue("Q_EXTERNALDB_ID",) best skip and parse sample information at sample level, no experiment-wide ID from what I can tell
+    # handle measured samples
+    for (barcode, datamap) in rawDataPerSample:
+        newLogFolder = copyLogs(currentPath, measuremnt.getLogFiles())
+        handleSingleSample(transaction, space, barcode, datamap, runExperiment, currentPath, newLogFolder)
+
+def handleSingleSample(transaction, space, parentSampleCode, mapWithDataForSample, openbisExperiment, currentPath, absLogPath):
```

Comment:
```suggestion
def createSample(transaction, space, parentSampleCode, mapWithDataForSample, openbisExperiment, currentPath, absLogPath):
```

Reply:
```suggestion
def createSampleWithData(transaction, space, parentSampleCode, mapWithDataForSample, openbisExperiment, currentPath, absLogPath):
```

Reply:
since more is happening there I think we can be even more explicit. also not to confuse it with the "createNewSample" method

Reply:
Sure, hit me :)